vllm/distributed/device_communicators/all2all.py

```diff
@@ -0,0 +1,80 @@
+# SPDX-License-Identifier: Apache-2.0
+import torch
+
+from vllm.forward_context import get_forward_context
+
+
+class All2AllBase:
+
+    def __init__(self, cpu_group, model):
```

Comment:
for how to use `cpu_group` to initialize pplx, see https://github.com/ppl-ai/pplx-kernels/issues/18 as an example.

here we have access to the model instance, so we can also get moe configs here, and we can assume all moe layers have the same config, without `_all_to_all_cache` .

Reply:
from @nandor , 

```python
uid = nvshmem_get_unique_id() if rank == 0 else nvshmem_alloc_empty_unique_id()
torch.distributed.broadcast(uid, src=0)
nvshmem_init(uid, rank, world_size)
```

is not necessary for intranode code.

we can tell if we are in single node with https://github.com/vllm-project/vllm/blob/79a1d25bbd09420ad7d8631671a70cbcaf496cfb/vllm/distributed/parallel_state.py#L1127